File: src/firefly/java/edu/caltech/ipac/astro/net/Resolver.java
/*
 * License information at https://github.com/Caltech-IPAC/firefly/blob/master/License.txt
 */
package edu.caltech.ipac.astro.net;
/**
 * User: roby
 * Date: Mar 4, 2009
 */
import java.io.Serializable;

/**
 * @author Trey Roby
 */
public enum Resolver implements Serializable {
    NED("ned", "NED", (Resolver) null),
    Simbad("simbad", "Simbad", (Resolver) null),
    NedThenSimbad("nedthensimbad", "Try NED then Simbad", NED, Simbad),
    SimbadThenNed("simbadthenned", "Try Simbad then NED", Simbad, NED),
    PTF("ptf", "PTF", (Resolver) null),
    UNKNOWN("unknown", "resolved with unknown resolver", (Resolver) null),
    NONE("none", "None", (Resolver) null);

    private final String _key;
    private final String _desc;
    private final Resolver[] _concreteResolvers;

    Resolver(String key, String desc, Resolver... concreteResolvers) {
        _key = key;
        _desc = desc;
        _concreteResolvers = concreteResolvers;
    }

    public String toString() { return _key; }
    public Resolver[] getConcertResolvers() { return _concreteResolvers !=null ? _concreteResolvers : new Resolver[] {this}; }

    public static Resolver parse(String resolveStr) {
        if (resolveStr.equalsIgnoreCase("ned")) return Resolver.NED;
        else if (resolveStr.equalsIgnoreCase("simbad")) return Resolver.Simbad;
        else if (resolveStr.equalsIgnoreCase("nedthensimbad")) return Resolver.NedThenSimbad;
        else if (resolveStr.equalsIgnoreCase("simbadthenned")) return Resolver.SimbadThenNed;
        else if (resolveStr.equalsIgnoreCase("ptf")) return Resolver.PTF;
        else return Resolver.NONE;
    }
}

File: src/firefly/js/ipac-firefly/astro/net/Resolver.js
/*jshint browserify:true*/
/*jshint curly:false*/
"use strict";

var Resolver=  {
    NED           : {key : "ned", desc : "NED"},
    Simbad        : {key : "simbad", desc : "simbad"},
    NedThenSimbad : {key : "nedthensimbad", desc : "try ned then simbad" },
    SimbadThenNed : {key : "simbadthenned", value : "try simbad then ned"},
    PTF           : {key : "ptf", desc : "ptf"},
    smart         : {key : "smart", desc  : "try ned and simbad, then decide"},
    UNKNOWN       : {key : "unknown", desc : "resolved with unknown resolver"},
    NONE          : {key : "none", desc : "none"}
};

Resolver.NedThenSimbad.combination=  [Resolver.NED, Resolver.Simbad];
Resolver.SimbadThenNed.combination=  [Resolver.Simbad, Resolver.NED];

function toString() {
    return this.key;
}

Resolver.NED.toString= toString;
Resolver.Simbad.toString= toString;
Resolver.NedThenSimbad.toString= toString;
Resolver.SimbadThenNed.toString= toString;
Resolver.PTF.toString= toString;
Resolver.smart.toString= toString;
Resolver.UNKNOWN.toString= toString;
Resolver.NONE.toString= toString;



var parse= function(resolveStr) {

    if (!resolveStr) return null;
    var retval= null;
    for( var resolveType in Resolver) {
        if( Resolver.hasOwnProperty( resolveType ) ) {
            if (resolveStr.toLowerCase()===Resolver[resolveType].key) {
                retval= Resolver[resolveType];
                break;
            }
        }
    }
    return retval;
};

exports.NED= Resolver.NED;
exports.simbad= Resolver.simbad;
exports.nedthensimbad= Resolver.nedthensimbad;
exports.simbadthenned= Resolver.simbadthenned;
exports.ptf= Resolver.ptf;
exports.smart= Resolver.smart;
exports.unknown= Resolver.unknown;
exports.none= Resolver.none;
exports.parse= parse;
